src/test-e2e/applications/applications.po.ts

```diff
@@ -15,6 +16,25 @@ export class ApplicationsPage extends CFPage {
 
   appList = new ListComponent();
 
+  static goToAppSummary(appName: string, cfGuid: string, appGuid: string): ApplicationPageSummaryTab {
+    const appsPage = new ApplicationsPage();
+    // Find app card
+    appsPage.sideNav.goto(SideNavMenuItem.Applications);
+    appsPage.appList.header.setSearchText(appName);
+
+    // Check for single card
+    expect(appsPage.appList.header.getSearchText()).toEqual(appName);
+    expect(appsPage.appList.cards.getCardCount()).toBe(1);
+    const cardP = appsPage.appList.cards.findCardByTitle(appName);
+
+    // Go to summary
+    protractor.promise.controlFlow().execute(() => cardP.then(card => card.click()));
```

Comment:
I don't think you need to use controlFlow - this is confusing - I think you can just use appsPage.appList.cards.findCardByTitle(appName).then - this will run in the control flow.